src/snowflake/connector/constants.py

```diff
@@ -95,6 +95,14 @@ class FieldType(NamedTuple):
     FieldType(
         name="GEOMETRY", dbapi_type=[DBAPI_TYPE_STRING], pa_type=lambda: pa.string()
     ),
+    FieldType(
+        # TODO(SNOW-969160): While pa.binary() results in the correct pandas column
+        # type being generated, it should be switched to pa.list_(...) once parsing
+        # for the new result metadata fields is added.
+        name="VECTOR",
+        dbapi_type=[DBAPI_TYPE_BINARY],
+        pa_type=lambda: pa.binary(),
```

Comment:
How would a caller convert this binary value back into a list of floats? Do we need to document endianness here for users to unpack this correctly?

Reply:
This type is only used when we're generating an empty pandas dataframe (to infer the type of the pandas columns). When rows are returned, we can infer the type from the payload itself. I have a test which verifies this pandas handling.

Reply:
Sounds good.